### ISSUE TITLE
tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.

### ISSUE THREAD

**Issue by Fchaubard** [tensorflow/tensorflow#66750]
### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.13.1

### Custom code

No

### OS platform and distribution

Ubuntu 20.04.6

### Python version

3.8.10

### GCC/compiler version

9.4.0

### CUDA/cuDNN version

CUDA 12.4

### GPU model and memory

GEFORCE GTX 4090

### Current behavior?

tensorflow does not detect the CUDA drivers but they are there and added to PATH.
---
python3 -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
2024-04-30 16:29:17.328463: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-04-30 16:29:17.362495: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-04-30 16:29:17.362915: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-04-30 16:29:18.094331: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-04-30 16:29:18.749023: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1960] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[]
---
nvidia-smi returns this suggesting the drivers are there and working:

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.78                 Driver Version: 550.78         CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090        Off |   00000000:03:00.0 Off |                  Off |
|  0%   36C    P8             12W /  450W |      29MiB /  24564MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      1089      G   /usr/lib/xorg/Xorg                              9MiB |
|    0   N/A  N/A      1142      G   /usr/bin/gnome-shell                            8MiB |
+-----------------------------------------------------------------------------------------+

Even pytorch recognizes the drivers without issue: 
---
Python 3.8.10 (default, Nov 22 2023, 10:22:35) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.cuda.device_count()
1
>>> torch.cuda.get_device_name(0)
'NVIDIA GeForce RTX 4090'
---

But not tensorflow? Please help! 

### Standalone code to reproduce the issue

```shell
You will need to recreate the environment and then:
pip install tensorflow
```

**Comment by tilakrayal** [comment]
@Fchaubard,
I suspect you are trying to install the every tensorflow version on CUDA 12.4 only which is not compatible. Could you please uninstall the installed libraries and try to install  Clang 16.0.0, Bazel 5.3.0, cuDNN - 8.6,	CUDA - 11.8  for the tensorflow v2.13 and check the below process.

```python
- Make sure cuda and cudnn are installed correctly.
- Make sure $Env:CUDA_PATH is given correctly
- Make sure $Env:LD_LIBRARY_PATH is given correctly
- Make sure $Env:TF_CUDA_PATHS is given correctly
```
https://www.tensorflow.org/install/source#gpu


Thank you!

**Comment by kiki81524** [comment]
`import tensorflow as tf
print("Num GPUs Available: ", len(tf.config.list_physical_devices('GPU')))`
And I get:
`2024-06-28 00:03:52.491725: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-06-28 00:03:52.493638: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-28 00:03:52.528877: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-28 00:03:52.529758: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-28 00:03:53.230172: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT

Num GPUs Available:  0

2024-06-28 00:03:53.984823: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1960] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...`

`import tensorflow as tf
tf.test.is_gpu_available()`
get:
`WARNING:tensorflow:From /tmp/ipykernel_1391302/2674676428.py:3: is_gpu_available (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.config.list_physical_devices('GPU')` instead.

2024-06-28 00:04:14.961278: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1960] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...

False`

I do not have sudo privileges on the GPU server I am using, so I cannot uninstall and reinstall CUDA and NVIDIA.
I can run GPU normally with Torch, but TensorFlow cannot recognize my GPU. I am using Python 3.8.18, tensorflow 2.13.1 and the GPU server is Ubuntu 20.04.6 LTS 64-bit.


The package version information, nvcc -V, and nvidia-smi information are attached.
`(copy_py38) kiki@RTX4090:~/Other_method/Pix2Pix$ nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Fri_Jan__6_16:45:21_PST_2023
Cuda compilation tools, release 12.0, V12.0.140
Build cuda_12.0.r12.0/compiler.32267302_0
(copy_py38) kiki@RTX4090:~/Other_method/Pix2Pix$ nvidia-smi
Fri Jun 28 00:05:05 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.171.04             Driver Version: 535.171.04   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090        Off | 00000000:65:00.0 Off |                  Off |
| 66%   74C    P2             314W / 450W |  19218MiB / 24564MiB |    100%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 4090        Off | 00000000:B3:00.0 Off |                  Off |
|  0%   49C    P8              23W / 450W |    789MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1073      G   /usr/lib/xorg/Xorg                           46MiB |
|    0   N/A  N/A      1335      G   /usr/bin/gnome-shell                         12MiB |
|    0   N/A  N/A   1379277      C   ...anaconda3/envs/copy_py38/bin/python      384MiB |
|    0   N/A  N/A   1384198      C   python                                    18368MiB |
|    0   N/A  N/A   1391302      C   ...anaconda3/envs/copy_py38/bin/python      384MiB |
|    1   N/A  N/A      1073      G   /usr/lib/xorg/Xorg                            4MiB |
|    1   N/A  N/A   1379277      C   ...anaconda3/envs/copy_py38/bin/python      384MiB |
|    1   N/A  N/A   1391302      C   ...anaconda3/envs/copy_py38/bin/python      384MiB |
+---------------------------------------------------------------------------------------+`

![tensorflow無法使用01](https://github.com/tensorflow/tensorflow/assets/167969545/31ee6ac1-e5f5-425f-bfe6-0706a560462e)
![tensorflow無法使用02](https://github.com/tensorflow/tensorflow/assets/167969545/dbc46eaa-ffa5-43c4-a1cf-f53549066fe9)
`Package                      Version
---------------------------- -----------
absl-py                      2.1.0
anyio                        3.7.1
argon2-cffi                  23.1.0
argon2-cffi-bindings         21.2.0
asttokens                    2.0.5
astunparse                   1.6.3
attrs                        23.2.0
Babel                        2.14.0
backcall                     0.2.0
beautifulsoup4               4.12.3
bleach                       6.1.0
cachetools                   4.2.4
certifi                      2024.2.2
cffi                         1.16.0
charset-normalizer           3.3.2
click                        8.1.7
clip                         1.0
cmake                        3.28.3
coloredlogs                  15.0.1
colormath                    3.0.0
comm                         0.2.1
configparser                 6.0.1
contourpy                    1.1.1
cssutils                     2.11.0
cycler                       0.12.1
debugpy                      1.6.7
decorator                    4.4.2
defusedxml                   0.7.1
diffvg                       0.0.1
docker-pycreds               0.4.0
entrypoints                  0.4
exceptiongroup               1.2.0
executing                    0.8.3
fastjsonschema               2.19.1
ffmpeg                       1.4
filelock                     3.13.1
flatbuffers                  24.3.25
fonttools                    4.51.0
fsspec                       2024.3.0
ftfy                         6.2.0
gast                         0.3.3
gdown                        4.6.0
gitdb                        4.0.11
GitPython                    3.1.42
google-auth                  2.30.0
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
greenlet                     3.0.3
grpcio                       1.63.0
h5py                         2.10.0
humanfriendly                10.0
idna                         3.6
imageio                      2.34.1
imageio-ffmpeg               0.4.9
importlib-metadata           4.6.4
importlib_resources          6.4.0
ipykernel                    6.1.0
ipython                      7.26.0
ipython-genutils             0.2.0
ipywidgets                   8.1.2
jedi                         0.18.1
Jinja2                       3.1.3
joblib                       1.3.2
json5                        0.9.5
jsonpatch                    1.33
jsonpointer                  2.4
jsonschema                   3.2.0
jupyter-client               6.1.12
jupyter-core                 4.7.1
jupyter-server               1.10.2
jupyterlab                   3.1.6
jupyterlab-pygments          0.1.2
jupyterlab-server            2.7.0
jupyterlab_widgets           3.0.10
keras                        2.13.1
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.5
libclang                     18.1.1
lit                          18.1.4
llvmlite                     0.41.1
lpips                        0.1.4
Markdown                     3.6
MarkupSafe                   2.1.5
matplotlib                   3.7.5
matplotlib-inline            0.1.2
mistune                      3.0.2
moviepy                      1.0.3
mpmath                       1.3.0
natsort                      8.4.0
nbclassic                    0.5.6
nbclient                     0.7.0
nbconvert                    7.16.2
nbformat                     5.10.3
nest-asyncio                 1.6.0
networkx                     3.1
ninja                        1.11.1.1
notebook                     6.4.3
notebook_shim                0.2.4
numba                        0.58.1
numpy                        1.24.3
nvidia-cublas-cu11           11.10.3.66
nvidia-cublas-cu12           12.1.3.1
nvidia-cuda-cupti-cu11       11.7.101
nvidia-cuda-cupti-cu12       12.1.105
nvidia-cuda-nvrtc-cu11       11.7.99
nvidia-cuda-nvrtc-cu12       12.1.105
nvidia-cuda-runtime-cu11     11.7.99
nvidia-cuda-runtime-cu12     12.1.105
nvidia-cudnn-cu11            8.5.0.96
nvidia-cudnn-cu12            8.9.2.26
nvidia-cufft-cu11            10.9.0.58
nvidia-cufft-cu12            11.0.2.54
nvidia-curand-cu11           10.2.10.91
nvidia-curand-cu12           10.3.2.106
nvidia-cusolver-cu11         11.4.0.1
nvidia-cusolver-cu12         11.4.5.107
nvidia-cusparse-cu11         11.7.4.91
nvidia-cusparse-cu12         12.1.0.106
nvidia-ml-py3                7.352.0
nvidia-nccl-cu11             2.14.3
nvidia-nvjitlink-cu12        12.4.99
nvidia-nvtx-cu11             11.7.91
nvidia-nvtx-cu12             12.1.105
oauthlib                     3.2.2
opencv-python                4.5.3.56
opt-einsum                   3.3.0
packaging                    23.2
pandas                       2.0.3
pandocfilters                1.5.1
parso                        0.8.3
pathtools                    0.1.2
pexpect                      4.8.0
pickleshare                  0.7.5
pillow                       10.3.0
pip                          24.0
platformdirs                 3.10.0
plotly                       5.2.1
proglog                      0.1.10
prometheus_client            0.20.0
promise                      2.3
prompt-toolkit               3.0.43
protobuf                     4.25.3
psutil                       5.8.0
ptyprocess                   0.7.0
pure-eval                    0.2.2
pyaml                        24.4.0
pyasn1                       0.6.0
pyasn1_modules               0.4.0
pycparser                    2.21
Pygments                     2.15.1
pyparsing                    3.1.2
pyrsistent                   0.20.0
PySocks                      1.7.1
python-dateutil              2.9.0.post0
pytz                         2024.1
PyWavelets                   1.4.1
PyYAML                       6.0.1
pyzmq                        25.1.2
regex                        2024.5.15
requests                     2.31.0
requests-oauthlib            2.0.0
requests-unixsocket          0.3.0
rsa                          4.9
scikit-image                 0.18.1
scikit-learn                 1.0.2
scipy                        1.4.1
seaborn                      0.13.2
Send2Trash                   1.8.2
sentry-sdk                   1.42.0
setuptools                   68.2.2
shortuuid                    1.0.13
six                          1.16.0
smmap                        5.0.1
sniffio                      1.3.1
soupsieve                    2.5
SQLAlchemy                   2.0.28
stack-data                   0.2.0
subprocess32                 3.5.4
svgpathtools                 1.6.1
svgwrite                     1.4.3
sympy                        1.12
tenacity                     8.2.3
tensorboard                  2.13.0
tensorboard-data-server      0.7.2
tensorboard-plugin-wit       1.8.1
tensorflow                   2.13.1
tensorflow-estimator         2.13.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    2.4.0
terminado                    0.18.1
threadpoolctl                3.3.0
tifffile                     2023.7.10
tinycss2                     1.2.1
torch                        2.0.1
torch-tools                  0.1.5
torchfile                    0.1.0
torchvision                  0.15.2
tornado                      6.4.1
tqdm                         4.66.2
traitlets                    5.14.2
triton                       2.0.0
typing_extensions            4.5.0
tzdata                       2024.1
urllib3                      2.2.1
visdom                       0.2.4
wandb                        0.12.0
wcwidth                      0.2.13
webencodings                 0.5.1
websocket-client             0.57.0
Werkzeug                     3.0.3
wheel                        0.41.2
widgetsnbextension           4.0.10
wrapt                        1.16.0
zipp                         3.17.0`

**Comment by bdbaigc** [comment]
i solved my problem by install tensorflow by conda，commands as follows：
conda create -n tf-gpu tensorflow-gpu
conda activate tf-gpu

refrence: https://docs.anaconda.com/working-with-conda/applications/tensorflow/

**Comment by woshinigange** [comment]
> i solved my problem by install tensorflow by conda，commands as follows： conda create -n tf-gpu tensorflow-gpu conda activate tf-gpu
> 
> refrence: https://docs.anaconda.com/working-with-conda/applications/tensorflow/

The current setup installs TensorFlow version 1.0, but if you want to use version 2.0, this setup will not work.
